experimental/LibSingWrapper.jl

```diff
@@ -0,0 +1,244 @@
+
+  # compute the jacobian ideal
+  Df = jacobi_matrix(f)::AbstractAlgebra.Generic.MatSpaceElem{typeof(f)}
+  I = ideal(R, [Df[i, 1] for i in 1:nrows(Df)])
+  
+  # transfer to the localized ring since otherwise we could not work with 
+  # local orderings
+  WI = W(I)
+  # compute a standard basis w.r.t a local ordering 
+  stdI = groebner_basis(WI)
+
+  # carry out the computations on the Singular side
+  Singular.dimension(singular_gens(stdI)) == 0 || error("singularity is not isolated")
+  milnor_number = Singular.vdim(singular_gens(stdI)) 
+  # make the conversion to the Oscar side 
+  milnor_kbase = [to_oscar_side(stdI, g) for g in gens(Singular.kbase(singular_gens(stdI)))]
+  return (stdI, milnor_number, milnor_kbase)
+end
+
+### The Tjurina number  of an isolated hypersurface singularity f
+@Markdown.doc """
+    tjurina(f::MPolyElem; point::Vector=[])
+
+For an isolated hypersurface singularity `f` at a point `point`,
+compute the triple `(gb,τ,g)` where `gb` is a standard basis for the
+Tjurina ideal. `τ` is the Tjurina number, and `g` is a ``k``-base of the
+Tjurina algebra.
+
+The default argument for `point` is the origin.
+"""
+function tjurina(f::MPolyElem; point::Vector{T}=Vector{elem_type(base_ring(parent(f)))}()) where {T}
+  iszero(f) && error("input polynomial is zero")
+  R = parent(f)
+  kk = coefficient_ring(f)
+  all((x->(parent(x) == kk)), point) || return tjurina(f, point=kk.(point))
+
+  # Set sanity check on point or creation of vector of point, if default point 
+  k_point = kk.(point)
+  if length(point) != 0
+     # right number of coordinates?
+     length(k_point) == ngens(R) || error("vector does not have correct length")
+  else
+     # default case, need to set up point for origin
+     k_point = elem_type(kk)[zero(kk) for i in 1:ngens(R)]
+  end
+
+  # localize at point (setting up stage for Singular)
+  RL = Localization(MPolyComplementOfKPointIdeal(R,k_point))  
+
+  # compute the jacobian and tjurina ideals
+  Jf = jacobi_matrix(f)::AbstractAlgebra.Generic.MatSpaceElem{typeof(f)}
+  I=ideal(R,[Jf[i,1] for i in 1:nrows(Jf)]) + ideal(R,[f])
+  # now move to localization RL of R and do standard basis there
+  IL=RL(I)
+  stdIL = groebner_basis(IL)
+  
+  # now fill in the return values
+  Singular.dimension(singular_gens(stdIL)) == 0 || error("singularity is not isolated")
+  tjurina_number = Singular.vdim(singular_gens(stdIL)) 
+  # make the conversion to the Oscar side 
+  tjurina_kbase = [to_oscar_side(stdIL, g) for g in gens(Singular.kbase(singular_gens(stdIL)))]
+  return (stdIL, tjurina_number, tjurina_kbase)
+end
+
+### The global Milnor number of a hypersurface with at most isolated singularities
+@Markdown.doc """
+    global_milnor_number(f::MPolyElem)
+
+Computes the global Milnor number of the affine hypersurface 
+defined by `f` with at most isolated singularities.
+"""
+function global_milnor_number(f::MPolyElem{T}) where {T<:FieldElem}
```

Comment:
Your choice, but looking at the names you should have
 - tjurina -> the number + data
 - tjurina_number
 - tjurina_number_global (then tab-expansion will find it)